### PR TITLE
Fix absolute path for test on macos

### LINKED ISSUE #3797: Binary component create fails if path is relative and includes ../
/kind bug

## What versions of software are you using?

**Operating System:**
macOS
**Output of `odo version`:**
master
## How did you run odo exactly?
```
It("binary component is valid if path is relative and includes ../", func() {
       oc.ImportJavaIS(project)
       helper.CopyExample(filepath.Join("binary", "java", "openjdk"), context)

       relativeContext := fmt.Sprintf("..%c%s", filepath.Separator, filepath.Base(context))
       fmt.Printf("relativeContext = %#v\n", relativeContext)

      helper.CmdShouldPass("odo", append(args, "create", "java:8", "sb-jar-test", "--project",
             project, "--binary", filepath.Join(context, "sb.jar"), "--context", relativeContext)...)
      helper.ValidateLocalCmpExist(relativeContext, "Type,java:8", "Name,sb-jar-test")
})
```

## Actual behavior
```
[oc] 'Running odo with args [odo component create java:8 sb-jar-test --project hkanwitwml --binary 
/var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/640472875/sb.jar --context ../640472875]
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✗  Validating component [543ms]
[odo]  ✗  ../../../../../../../var/folders/c9/fb55pf_568dfd1fzj9w1v4n00000gn/T/640472875/sb.jar binary needs to be inside of the context directory (../640472875)
```
## Expected behavior
Should pass

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this?**
/kind failing-test

**What does does this PR do / why we need it**:
This PR will fix mac context related issue for test. 
**Which issue(s) this PR fixes**:

Fixes #3797

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: tests/integration/component.go [hunk 1]
```diff
@@ -308,8 +308,13 @@ func componentTests(args ...string) {
 			relativeContext := fmt.Sprintf("..%c%s", filepath.Separator, filepath.Base(commonVar.Context))
 			fmt.Printf("relativeContext = %#v\n", relativeContext)
 
-			helper.CmdShouldPass("odo", append(args, "create", "--s2i", "java:8", "sb-jar-test", "--project",
-				commonVar.Project, "--binary", filepath.Join(commonVar.Context, "sb.jar"), "--context", relativeContext)...)
+			if runtime.GOOS == "darwin" {
```
Comment (girishramnani) [inline tests/integration/component.go:311]: whats the difference between these two conditions?

Comment (girishramnani) [reply tests/integration/component.go:311]:
```
helper.CmdShouldPass("odo", append(args, "create", "--s2i", "java:8", "sb-jar-test", "--project",
					commonVar.Project, "--binary", filepath.Join(commonVar.Context, "sb.jar"), "--context", relativeContext)...)
```

and 
```
helper.CmdShouldPass("odo", append(args, "create", "--s2i", "java:8", "sb-jar-test", "--project",
					commonVar.Project, "--binary", filepath.Join(commonVar.Context, "sb.jar"), "--context", relativeContext)...)
```

Comment (anandrkskd) [reply tests/integration/component.go:311]: For mac some how the test are using /private/var/.../afaf(random dir) to run test which is not the case for linux and windows, Changing the context passing according to working dir w.r.t test solves the issue.

Comment (girishramnani) [reply tests/integration/component.go:311]: sorry but before there wasn't a "/private` in the filepath.Join, and I would suggest not force pushing the code

Comment (anandrkskd) [reply tests/integration/component.go:311]: Understood

Comment (girishramnani) [reply tests/integration/component.go:311]: /approve

**Comment by prietyc123** [comment]
I verified the change locally on macOS and the test runs successfully.

/lgtm

**Pull request merged**